CoreFoundation/Preferences.subproj/CFPreferences.c

```diff
@@ -440,7 +440,11 @@ static CFURLRef _CFPreferencesURLForStandardDomainWithSafetyLevel(CFStringRef do
     CFURLRef theURL = NULL;
     CFAllocatorRef prefAlloc = __CFPreferencesAllocator();
 #if DEPLOYMENT_TARGET_MACOSX || DEPLOYMENT_TARGET_LINUX || DEPLOYMENT_TARGET_WINDOWS
+#ifdef __ANDROID__
+    CFURLRef prefDir = CFBundleCopyBundleURL(CFBundleGetMainBundle());
```

Comment:
It is not correct to write from or to the main bundle. The system must function if the application is running with its main bundle path read only.

This should be a location where application-specific user information can be read or written to. A quick search shows the closest would be the `internalDataPath` vended through a `ANativeActivity`, but that may not be the layering you want. Either way, this is incorrect in the face of the semantics of this API.

Reply:
If you can accept using the cacheDirectory as the bundle directory then this follows rather than replicate the dependency on various environment variables.

Reply:
The semantics of a bundle directory is that it, loosely, is where the executable is. I assume that is not writable on Android, and the API contract for CFBundle is that bundles may be read-only. No, this doesn't work.

What is a correct path for storing _user_-specific data on Android? Note that on FHS systems we are falling back to `$HOME/.local` (per [XDG](https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.6.html)); is that good on Android as well, or is there a better Android equivalent?

Reply:
The only writable directory known to an android app is  the “cacheDir”.